Sprint 07

Peppa-Akk opened the pull request:
Create unit and UI tests;
Refactoring the project.

Peppa-Akk committed:
Ended 5-th sprint

Peppa-Akk committed:
Ended 6-th sprint

Peppa-Akk committed:
Ended 7-th sprint:

- Create unit and UI tests;
- Refactoring the project

amardanov commented on MovieQuiz/Models/QuizQuestion.swift:
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
```
Лишние пустые строчки лучше не оставлять, чтобы код выглядел аккуратнее и чище

amardanov commented on MovieQuiz/Models/QuizResultsViewModel.swift:
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+
```
Также по пустым строчкам.
И еще тут немного поехали отступы на 5,6,7 строчках. Можно исправить с помощью команд `cmd+a` + `control + i`

amardanov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,139 @@
+        viewController.showLoadingIndicator()
+        questionFactory?.loadData()
+    }
+
```
Лишняя пустая строчка

amardanov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    func proceedToNextQuestionOrResults() {
+        if self.isLastQuestion() {
```
Здесь и на 134 строчке можно не писать `self`, так как компилятор итак поймет к какому методу надо обращаться

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,176 +1,89 @@
 }
-
 /*
  Mock-данные
```
Mock данные теперь тоже кажется уже не нужны. Поэтому их можно удалить, тем самым почистится код

amardanov commented on MovieQuiz/Services/QuestionFactory.swift:
```diff
@@ -0,0 +1,103 @@
+}
+
+
+//    private let questions: [QuizQuestion] = [
```
Такие комментарии с кодом тоже не стоит оставлять ради чистоты кода

amardanov commented on MovieQuizTests/MovieQuizTests.swift:
```diff
@@ -0,0 +1,30 @@
+import XCTest
+
+struct ArithmeticOperations {
```
Этот файл есть в файловой системе, но почему-то в проекте не отображается. Возможно стоит его добавить вручную или удалить из папки проекта в Finder

amardanov commented on MovieQuizUITests/MovieQuizUITestsLaunchTests.swift:
```diff
@@ -0,0 +1,32 @@
+
+import XCTest
+
+final class MovieQuizUITestsLaunchTests: XCTestCase {
```
Такая же история как и с файлом `MovieQuizTests.swift`

Peppa-Akk committed:
Fix:
- Add protocol for AlertPresenter
- Correct for MVP
- Unnecessary indentes are removed
